Cannot build docker image in 2.7.1 from the source tarball

During the validation of 2.7.1rc1 we discovered that it is not possible to build the docker image from the source tarball.

The problem is that we are using "git" command outside of a git repo.

This is the error:
```
Executing command line:
[/Users/enrico.olivelli/Downloads/pulsar271/pulsar-2.7.1-candidate-1/apache-pulsar-2.7.1/docker/pulsar/../../pulsar-client-cpp/docker/build-wheels.sh,
3.7 cp37-cp37m]

fatal: not a git repository (or any of the parent directories): .git

[INFO] --- exec-maven-plugin:1.6.0:exec
(build-pulsar-clients-python-35) @ pulsar-docker-image ---

fatal: not a git repository (or any of the parent directories): .git

[ERROR] Command execution failed.

org.apache.commons.exec.ExecuteException: Process exited with an
error: 128 (Exit value: 128)

    at org.apache.commons.exec.DefaultExecutor.executeInternal
(DefaultExecutor.java:404)

    at org.apache.commons.exec.DefaultExecutor.execute
(DefaultExecutor.java:166)

    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)

    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)

    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)

    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo
(DefaultBuildPluginManager.java:137)

    at org.apache.maven.lifecycle.internal.MojoExecutor.execute
(MojoExecutor.java:210)

    at org.apache.maven.lifecycle.internal.MojoExecutor.execute
(MojoExecutor.java:156)

    at org.apache.maven.lifecycle.internal.MojoExecutor.execute
(MojoExecutor.java:148)

    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject
(LifecycleModuleBuilder.java:117)

    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject
(LifecycleModuleBuilder.java:81)

    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build
(SingleThreadedBuilder.java:56)

    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute
(LifecycleStarter.java:128)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)

    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)

    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)

    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke
(NativeMethodAccessorImpl.java:62)

    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke
(DelegatingMethodAccessorImpl.java:43)

    at java.lang.reflect.Method.invoke (Method.java:566)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced
(Launcher.java:282)

    at org.codehaus.plexus.classworlds.launcher.Launcher.launch
(Launcher.java:225)

    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode
(Launcher.java:406)

    at org.codehaus.plexus.classworlds.launcher.Launcher.main
(Launcher.java:347)
```

--- a/pulsar-client-cpp/docker/build-wheels.sh
+++ b/pulsar-client-cpp/docker/build-wheels.sh
@@ -23,7 +23,7 @@ set -e
 
 BUILD_IMAGE_NAME="${BUILD_IMAGE_NAME:-apachepulsar/pulsar-build}"
 
-ROOT_DIR=$(git rev-parse --show-toplevel)
+ROOT_DIR=$(dirname $0)/../..
 cd $ROOT_DIR
 
 PYTHON_VERSIONS=(

--- a/pulsar-client-cpp/pkg/deb/build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/build-deb.sh
@@ -21,7 +21,7 @@
 set -e -x
 
 cd /pulsar
-SRC_ROOT_DIR=$(git rev-parse --show-toplevel)
+SRC_ROOT_DIR=$(dirname $0)/../..
 cd $SRC_ROOT_DIR/pulsar-client-cpp/pkg/deb
 
 POM_VERSION=`$SRC_ROOT_DIR/src/get-project-version.py`

--- a/pulsar-client-cpp/pkg/deb/build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/build-deb.sh
@@ -21,7 +21,7 @@
 set -e -x
 
 cd /pulsar
-SRC_ROOT_DIR=$(dirname $0)/../..
+SRC_ROOT_DIR=$(pwd)
 cd $SRC_ROOT_DIR/pulsar-client-cpp/pkg/deb
 
 POM_VERSION=`$SRC_ROOT_DIR/src/get-project-version.py`

--- a/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(git rev-parse --show-toplevel)
+ROOT_DIR=$(dirname $0)/../../..
 
 docker pull apachepulsar/pulsar-build:debian-9
 

--- a/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../../..
+ROOT_DIR=$(dirname $0)/../../../..
 
 docker pull apachepulsar/pulsar-build:debian-9
 

--- a/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../../../..
+ROOT_DIR=$(dirname $0)/../..
 
 docker pull apachepulsar/pulsar-build:debian-9
 

--- a/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
+++ b/pulsar-client-cpp/pkg/deb/docker-build-deb.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../..
+ROOT_DIR=$(dirname $0)/../../..
 
 docker pull apachepulsar/pulsar-build:debian-9
 

--- a/pulsar-client-cpp/pkg/rpm/build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/build-rpm.sh
@@ -21,7 +21,7 @@
 set -e
 
 cd /pulsar
-ROOT_DIR=$(git rev-parse --show-toplevel)
+ROOT_DIR=$(dirname $0)/../..
 cd $ROOT_DIR/pulsar-client-cpp/pkg/rpm
 
 POM_VERSION=`$ROOT_DIR/src/get-project-version.py`

--- a/pulsar-client-cpp/pkg/rpm/build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/build-rpm.sh
@@ -21,7 +21,7 @@
 set -e
 
 cd /pulsar
-ROOT_DIR=$(pwd)
+ROOT_DIR=$(dirname $0)/../..
 cd $ROOT_DIR/pulsar-client-cpp/pkg/rpm
 
 POM_VERSION=`$ROOT_DIR/src/get-project-version.py`

--- a/pulsar-client-cpp/pkg/rpm/build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/build-rpm.sh
@@ -21,7 +21,7 @@
 set -e
 
 cd /pulsar
-ROOT_DIR=$(dirname $0)/../..
+ROOT_DIR=$(pwd)
 cd $ROOT_DIR/pulsar-client-cpp/pkg/rpm
 
 POM_VERSION=`$ROOT_DIR/src/get-project-version.py`

--- a/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../..
+ROOT_DIR=$(dirname $0)/../../../..
 
 docker pull apachepulsar/pulsar-build:centos-7
 

--- a/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(git rev-parse --show-toplevel)
+ROOT_DIR=$(dirname $0)/../../..
 
 docker pull apachepulsar/pulsar-build:centos-7
 

--- a/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../../..
+ROOT_DIR=$(dirname $0)/../../../..
 
 docker pull apachepulsar/pulsar-build:centos-7
 

--- a/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
+++ b/pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
@@ -20,7 +20,7 @@
 
 set -e
 
-ROOT_DIR=$(dirname $0)/../../../..
+ROOT_DIR=$(dirname $0)/../..
 
 docker pull apachepulsar/pulsar-build:centos-7
 

